### PR TITLE
cel: use singleton builder in rate limit

### PR DESCRIPTION
**Pull request by kyessenov**
Commit Message: Follow up to https://github.com/envoyproxy/envoy/pull/27861, for CEL rate limit descriptor.
Additional Description: Requires signature change for descriptor factory:
```c++
   virtual DescriptorProducerPtr
   createDescriptorProducerFromProto(const Protobuf::Message& config,
-                                    ProtobufMessage::ValidationVisitor& validator) PURE;
+                                    Server::Configuration::CommonFactoryContext& context) PURE;
```
Risk Level: low
Testing: yes
Docs Changes: none
Release Notes: none

### CODE REVIEW

**Comment by kyessenov** [comment]
/retest
(all flakes)

File: test/per_file_coverage.sh [hunk 1]
```diff
@@ -49,8 +49,8 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/extensions/filters/network/thrift_proxy/filters/payload_to_metadata:96.3"
 "source/extensions/filters/network/wasm:76.9"
 "source/extensions/http/cache/simple_http_cache:95.9"
-"source/extensions/rate_limit_descriptors:95.5"
-"source/extensions/rate_limit_descriptors/expr:95.5"
+"source/extensions/rate_limit_descriptors:95.0"
+"source/extensions/rate_limit_descriptors/expr:95.0"
```
Comment (KBaichoo) [inline test/per_file_coverage.sh:53]: How feasible would it be to increase the coverage?

**Review by KBaichoo — COMMENTED** [review]
lgtm otherwise, thanks

File: test/per_file_coverage.sh (hunk 1 above)
Comment (kyessenov) [reply test/per_file_coverage.sh:53]: It's not possible, the missing lines are macro guards.

Comment (RyanTheOptimist) [reply test/per_file_coverage.sh:53]: Boy I *hate* the deficiencies in our coverage tooling which lead to this.

**Review by RyanTheOptimist — COMMENTED** [review]
test/per_file_coverage.sh LGTM

**Pull request merged**